pages/docs/reference/sealed-classes.md

```diff
@@ -7,38 +7,86 @@ title: "Sealed Classes"
+_Sealed_ classes represent restricted class hierarchies that provide more control over inheritance. 
+All subclasses of a sealed class are known at compile time. No other subclasses may appear after
+a module with the sealed class is compiled. For example, third-party clients can't extend your sealed class in their code.
+Thus, instances of sealed classes can have one of the types from a limited set closed for further extension.
 
-To declare a sealed class, you put the `sealed` modifier before the name of the class. A sealed class can have
-subclasses, but all of them must be declared in the same file as the sealed class itself. (Before Kotlin 1.1,
-the rules were even more strict: classes had to be nested inside the declaration of the sealed class).
+In some sense, sealed classes are similar to enum classes: the set of values
+for an enum type is also restricted, but each enum constant exists only as a _single instance_, whereas a subclass
+of a sealed class can have _multiple_ instances, each with its own state.
+
+To declare a sealed class, put the `sealed` modifier before its name.
 
 <div class="sample" markdown="1" theme="idea" data-highlight-only>
+
 ```kotlin
 sealed class Expr
+
 data class Const(val number: Double) : Expr()
 data class Sum(val e1: Expr, val e2: Expr) : Expr()
 object NotANumber : Expr()
 ```
 </div>
 
-(The example above uses one additional new feature of Kotlin 1.1: the possibility for data classes to extend other
-classes, including sealed classes.)
-
 A sealed class is [abstract](classes.html#abstract-classes) by itself, it cannot be instantiated directly and can have *abstract*{: .keyword } members.
 
 Sealed classes are not allowed to have non-*private*{: .keyword } constructors (their constructors are *private*{: .keyword } by default).
 
-Note that classes which extend subclasses of a sealed class (indirect inheritors) can be placed anywhere, not necessarily in
-the same file.
+## Sealed interfaces
+
+> Sealed interfaces are [Experimental](evolution/components-stability.html). They may be dropped or changed at any time.
+> Opt-in is required (see details [below](#experimental-status-of-sealed-interfaces-and-extended-inheritance-rules)). Use them only for evaluation purposes. We appreciate your feedback on them in [YouTrack](https://youtrack.jetbrains.com/issues/KT-42433).
+{:.note}
+
+Interfaces can be declared `sealed` as well as classes. The `sealed` modifier works on interfaces the same way:
+all implementations of a sealed interface are known at compile time. Once a module with a sealed interface is compiled,
+no new implementations can appear.
+
+<div class="sample" markdown="1" theme="idea" data-highlight-only>
+
+```kotlin
+sealed interface Expr
+
+sealed class MathExpr(): Expr
+
+data class Const(val number: Double) : MathExpr()
+data class Sum(val e1: Expr, val e2: Expr) : MathExpr()
+object NotANumber : Expr
+```
+</div>
+
+## Inheritance rules
```

Comment:
This structure seems strange:
* Inheritance rules with information only about location. Is it the only rule? Do we need to call it `Inheritance rules`?
* Only one child heading.

`Extended` – is it our official naming? If we extend only one rule about location, it seems strange. Maybe use another word, such as additional or experimental.

Reply:
Good catch, thanks. I'll rename the sections.
Regarding one child heading: I think it's fine here, because:
- the subsection extends the location topic, so it's strange to put it outside this section.
- it must be separated because it describes an "advanced/future" aspect of this topic.